riscv/csrs.h

```diff
@@ -78,6 +78,9 @@ class basic_csr_t: public csr_t {
     return val;
   }
 
+  void set_val(reg_t v) { val = v; }
+  reg_t get_val() { return val; }
+
```

Comment:
This breaks encapsulation and allows any client code to cause all kinds of shenanigans, violating lots of assumptions made throughout the CSR code.